File: engine/engine-model/src/main/scala/tools/packet/ui/RefreshTeleportRockMapList.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class RefreshTeleportRockMapList(private var _vips: java.util.List[java.lang.Integer], private var _regulars: java.util.List[java.lang.Integer], private var _delete: Boolean, private var _vip: Boolean) extends PacketInput {
  def vips: java.util.List[java.lang.Integer] = _vips

  def regulars: java.util.List[java.lang.Integer] = _regulars

  def delete: Boolean = _delete

  def vip: Boolean = _vip

  override def opcode(): SendOpcode = SendOpcode.MAP_TRANSFER_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/ui/RefreshTeleportRockMapList.java
package tools.packet.ui;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record RefreshTeleportRockMapList(List<Integer> vips, List<Integer> regulars, Boolean delete,
                                         Boolean vip) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MAP_TRANSFER_RESULT;
   }
}